src/UnitTests/GitHub.App/Services/PullRequestServiceTests.cs

```diff
@@ -17,6 +17,241 @@
+
+                Assert.False(isClean);
+            }
+        }
+
+        [Fact]
+        public async Task RemovedFile_False()
+        {
+            using (var tempDir = new TempDirectory())
+            using (var repo = CreateRepository(tempDir))
+            {
+                var service = CreatePullRequestService(repo);
+                var repositoryModel = CreateLocalRepositoryModel(repo);
+                var path = "file.txt";
+                var file = Path.Combine(repo.Info.WorkingDirectory, path);
+                File.WriteAllText(file, "contents");
+                Commands.Stage(repo, path);
+                repo.Commit("foo", Author, Author);
+                File.Delete(file);
+                Commands.Stage(repo, path);
+
+                var isClean = await service.IsWorkingDirectoryClean(repositoryModel).FirstAsync();
+
+                Assert.False(isClean);
+            }
+        }
+
+        [Fact]
+        public async Task RenamedInIndexFile_False()
+        {
+            using (var tempDir = new TempDirectory())
+            using (var repo = CreateRepository(tempDir))
+            {
+                var service = CreatePullRequestService(repo);
+                var repositoryModel = CreateLocalRepositoryModel(repo);
+                var path = "file.txt";
+                var renamedPath = "renamed.txt";
+                var file = Path.Combine(repo.Info.WorkingDirectory, path);
+                var renamedFile = Path.Combine(repo.Info.WorkingDirectory, renamedPath);
+                File.WriteAllText(file, "contents");
+                Commands.Stage(repo, path);
+                repo.Commit("foo", Author, Author);
+                File.Move(file, renamedFile);
+                Commands.Stage(repo, path);
+                Commands.Stage(repo, renamedPath);
+
+                var isClean = await service.IsWorkingDirectoryClean(repositoryModel).FirstAsync();
+
+                Assert.False(isClean);
+            }
+        }
+
+        [Fact]
+        public async Task RenamedInWorkingDirFile_False()
+        {
+            using (var tempDir = new TempDirectory())
+            using (var repo = CreateRepository(tempDir))
+            {
+                var service = CreatePullRequestService(repo);
+                var repositoryModel = CreateLocalRepositoryModel(repo);
+                var path = "file.txt";
+                var renamedPath = "renamed.txt";
+                var file = Path.Combine(repo.Info.WorkingDirectory, path);
+                var renamedFile = Path.Combine(repo.Info.WorkingDirectory, renamedPath);
+                File.WriteAllText(file, "contents");
+                Commands.Stage(repo, path);
+                repo.Commit("foo", Author, Author);
+                File.Move(file, renamedFile);
+                // NOTE: Renamed files appear as Missing and Untracked rather than RenamedInWorkingDir. Is this a bug?
```

Comment:
@ethomson I'm surprised that renaming a (committed) file didn't make it appear as `RenamedInWorkingDir`. It instead appears as `Missing` and `Untracked`. This also makes sense, but I wonder what `RenamedInWorkingDir` is actually for? Might this be a bug or am I missing something obvious?

Reply:
You need to query for working directory renames explicitly in your Status, they're not included by default.  Since they look at _every_ untracked file, they're particularly expensive to calculate and I wouldn't recommend it, tbqh.

We added it to libgit2/LibGit2Sharp for Visual Studio 2013's Team Explorer when working in the mode where it hid the stage.  (If you have no staging area, you can't calculate renames between the HEAD and the index, so it calculated unstaged renames in the working directory).

Note that this default (_not_ detecting renames in the working directory) matches the behavior of https://github.com/git/git where status only shows staged renames (and in fact has no option to enable unstaged renames in the working directory).

Reply:
Thanks for clearing up my confusion!